lib/tasks/gitlab/web_hook.rake

```diff
@@ -7,15 +7,23 @@ namespace :gitlab do
 
       projects = find_projects(namespace_path)
 
+      with_hook_scope = ProjectHook.where(url: web_hook_url)
+      with_hook_scope = with_hook_scope.where(project: projects) unless projects == Project # there is Rails bug
```

Comment:
The problem is:
```
Loading production environment (Rails 4.1.9)
irb(main):001:0> ProjectHook.where(url: 'asdf', project: Project).to_sql
=> "SELECT \"web_hooks\".* FROM \"web_hooks\"  WHERE \"web_hooks\".\"type\" IN ('ProjectHook') AND \"web_hooks\".\"url\" = 'asdf' AND \"web_hooks\".\"project_id\" = 0  ORDER BY \"web_hooks\".\"created_at\" DESC, \"web_hooks\".\"id\" DESC"
```
So, when the `Project` isn't scoped then request contains incorrect condition.
But when we're use the scoped `Project` like:
```
irb(main):006:0> ProjectHook.where(url: 'asdf', project: Project.all.reorder('')).to_sql
=> "SELECT \"web_hooks\".* FROM \"web_hooks\"  WHERE \"web_hooks\".\"type\" IN ('ProjectHook') AND \"web_hooks\".\"url\" = 'asdf' AND \"web_hooks\".\"project_id\" IN (SELECT \"projects\".\"id\" FROM \"projects\")  ORDER BY \"web_hooks\".\"created_at\" DESC, \"web_hooks\".\"id\" DESC"
```
then resulting request is pretty correct.

I can change the 11 line of my patch or can change the method which returns the unscoped `Project` for `Project.all.reorder('')`. How to do better?

P.S. I don't know is it bug or not. But I think it is non-obvious behavior.